test/Either/Either.js

```diff
@@ -2,15 +2,124 @@
+    },
+    value: x
+  };
+};
+
+Identity.of = Identity;
+
+//  IdentityArb :: Arbitrary a -> Arbitrary (Identity a)
+var IdentityArb = function(arb) {
+  return arb.smap(Identity, function(i) { return i.value; });
+};
+
+//  MaybeArb :: Arbitrary a -> Arbitrary (Maybe a)
+var MaybeArb = function(arb) {
+  return jsc.oneof(JustArb(arb), jsc.constant(S.Nothing()));
+};
+
+//  JustArb :: Arbitrary a -> Arbitrary (Maybe a)
+var JustArb = function(arb) {
+  return arb.smap(S.Just, function(m) { return m.value; }, R.toString);
+};
+
+//  EitherArb :: Arbitrary a -> Arbitrary b -> Arbitrary (Either a b)
+var EitherArb = function(lArb, rArb) {
+  return jsc.oneof(LeftArb(lArb), RightArb(rArb));
+};
+
+//  LeftArb :: Arbitrary a -> Arbitrary (Either a b)
+var LeftArb = function(arb) {
+  return arb.smap(S.Left, function(e) { return e.value; }, R.toString);
+};
+
+//  RightArb :: Arbitrary a -> Arbitrary (Either b a)
+var RightArb = function(arb) {
+  return arb.smap(S.Right, function(e) { return e.value; }, R.toString);
+};
+
+//  Compose :: Apply f, Apply g
+//          => { of: b -> f b } -> { of: c -> g c }
+//          -> f (g a) -> Compose f g a
+var Compose = function(F, G) {
+  var _Compose = function _Compose(x) {
+    return {
+      constructor: _Compose,
+      map: function(f) {
+        return _Compose(R.map(R.map(f), x));
+      },
+      ap: function(y) {
+        return _Compose(R.ap(R.map(R.ap, x), y.value));
+      },
+      equals: function(other) {
+        return R.equals(x, other.value);
+      },
+      value: x
+    };
+  };
+  _Compose.of = function(x) {
+    return _Compose(F.of(G.of(x)));
+  };
+  return _Compose;
+};
+
 describe('Either', function() {
 
   it('throws if called', function() {
     throws(function() { S.Either(); },
            errorEq(Error, 'Cannot instantiate Either'));
   });
 
+  describe('Traversable laws', function() {
+
+    it('satisfies naturality', function() {
+      var maybeToEither = S.maybeToEither(0);
+      jsc.assert(jsc.forall(EitherArb(jsc.integer, MaybeArb(jsc.string)), function(either) {
+        var lhs = maybeToEither(either.sequence(S.Maybe.of));
+        var rhs = either.map(maybeToEither).sequence(S.Either.of);
+        return lhs.equals(rhs);
+      }));
+    });
+
+    it('satisfies identity', function() {
+      jsc.assert(jsc.forall(EitherArb(jsc.integer, jsc.string), function(either) {
+        var lhs = either.map(Identity).sequence(Identity.of);
+        var rhs = Identity.of(either);
+        return lhs.equals(rhs);
+      }));
+    });
+
+    it('satisfies composition', function() {
+      jsc.assert(jsc.forall(EitherArb(jsc.string, IdentityArb(EitherArb(jsc.string, jsc.integer))), function(u) {
+        var C = Compose(Identity, S.Either);
+        var lhs = u.map(C).sequence(C.of);
+        var rhs = C(u.sequence(Identity.of).map(function(x) {
+          return x.sequence(S.Either.of);
+        }));
+        return lhs.equals(rhs);
+      }));
+    });
+
+  });
+
```

Comment:
These look fine too. It gets a little confusing because we're using `Either` as both a `Traversable` and one of the `Applicative`s in the naturality test, which we might be able to tidy up at some point by creating a more general `Applicative` arbitrary instance.